operator/src/main/java/dev/responsive/k8s/operator/OperatorMain.java

```diff
@@ -21,12 +21,94 @@
 import io.fabric8.kubernetes.client.utils.Serialization;
 import io.javaoperatorsdk.operator.Operator;
+import java.io.File;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.PropertiesConfiguration;
+import org.apache.commons.configuration2.builder.fluent.Configurations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class OperatorMain {
+  private static final Logger LOGGER = LoggerFactory.getLogger(OperatorMain.class);
+  private static final String API_KEY_CONFIG = "responsive.platform.api.key";
+  private static final String SECRET_CONFIG = "responsive.platform.api.secret";
+
   public static void main(String[] args) {
+
+    final Options options = getOptions();
+    final CommandLineParser parser = new DefaultParser();
+    final HelpFormatter formatter = new HelpFormatter();
+
+    CommandLine cmd;
+
+    try {
+      cmd = parser.parse(options, args);
+    } catch (ParseException e) {
+      LOGGER.error("Error parsing command line params: ", e);
+      formatter.printHelp("Operator", options);
+      System.exit(1);
+      return;
+    }
+
+    final String target = cmd.getOptionValue("server-url");
+    final String configFileLocation = cmd.getOptionValue("config-file");
+    final Configuration config = getConfigurations(configFileLocation);
+
+    if (!(config.containsKey(API_KEY_CONFIG) && config.containsKey(SECRET_CONFIG))) {
+      LOGGER.error("Couldn't find API Key or secret properties in config file {}. "
+              + "We expect both {} and {} properties to be present", configFileLocation,
+              API_KEY_CONFIG, SECRET_CONFIG);
+      System.exit(1);
+    }
+
+    final String apiKey = config.getString(API_KEY_CONFIG);
+    final String secret = config.getString(SECRET_CONFIG);
+
     final Operator operator = new Operator();
     Serialization.jsonMapper().registerModule(new Jdk8Module());
-    operator.register(new ResponsivePolicyReconciler(new ControllerGrpcClient(args[0])));
+    operator.register(new ResponsivePolicyReconciler(new ControllerGrpcClient(target,
+            apiKey, secret)));
     operator.start();
   }
+
+  private static Configuration getConfigurations(String fileName) {
+    Configurations configurations = new Configurations();
+    try {
+      return configurations.properties(new File(fileName));
+    } catch (Exception e) {
+      LOGGER.error("Error loading configuration properties: {}", e.getMessage());
+    }
+    return new PropertiesConfiguration();
+  }
+
+  private static Options getOptions() {
+    Options options = new Options();
+    Option serverOption = Option.builder("server-url")
```

Comment:
thoughts on putting this in the config file instead? I think that would make it a bit easier when interacting with the docker image for the operator (maybe?) 

also we may want to name it `controller.url` instead in case we eventually have other types of servers